sdk/keyvault/azure-keyvault-certificates-apis/azure/keyvault/certificates/_shared/_generated/v2016_10_01/aio/operations_async/_key_vault_client_operations_async.pyi

```diff
@@ -0,0 +1,74 @@
+#
+# NOTE: This dynamically typed stub was automatically generated by stubgen.
+
+from typing import Any, Optional
+
+class KeyVaultClientOperationsMixin:
+    async def create_key(self, vault_base_url: Any, key_name: Any, kty: Any, key_size: Optional[Any] = ..., key_ops: Optional[Any] = ..., key_attributes: Optional[Any] = ..., tags: Optional[Any] = ..., curve: Optional[Any] = ..., *, cls: Optional[Any] = ..., **kwargs: Any): ...
+    async def import_key(self, vault_base_url: Any, key_name: Any, key: Any, hsm: Optional[Any] = ..., key_attributes: Optional[Any] = ..., tags: Optional[Any] = ..., *, cls: Optional[Any] = ..., **kwargs: Any): ...
+    async def delete_key(self, vault_base_url: Any, key_name: Any, *, cls: Optional[Any] = ..., **kwargs: Any): ...
+    async def update_key(self, vault_base_url: Any, key_name: Any, key_version: Any, key_ops: Optional[Any] = ..., key_attributes: Optional[Any] = ..., tags: Optional[Any] = ..., *, cls: Optional[Any] = ..., **kwargs: Any): ...
+    async def get_key(self, vault_base_url: Any, key_name: Any, key_version: Any, *, cls: Optional[Any] = ..., **kwargs: Any): ...
+    def get_key_versions(self, vault_base_url: Any, key_name: Any, maxresults: Optional[Any] = ..., *, cls: Optional[Any] = ..., **kwargs: Any): ...
+    def get_keys(self, vault_base_url: Any, maxresults: Optional[Any] = ..., *, cls: Optional[Any] = ..., **kwargs: Any): ...
+    async def backup_key(self, vault_base_url: Any, key_name: Any, *, cls: Optional[Any] = ..., **kwargs: Any): ...
+    async def restore_key(self, vault_base_url: Any, key_bundle_backup: Any, *, cls: Optional[Any] = ..., **kwargs: Any): ...
+    async def encrypt(self, vault_base_url: Any, key_name: Any, key_version: Any, algorithm: Any, value: Any, *, cls: Optional[Any] = ..., **kwargs: Any): ...
+    async def decrypt(self, vault_base_url: Any, key_name: Any, key_version: Any, algorithm: Any, value: Any, *, cls: Optional[Any] = ..., **kwargs: Any): ...
+    async def sign(self, vault_base_url: Any, key_name: Any, key_version: Any, algorithm: Any, value: Any, *, cls: Optional[Any] = ..., **kwargs: Any): ...
+    async def verify(self, vault_base_url: Any, key_name: Any, key_version: Any, algorithm: Any, digest: Any, signature: Any, *, cls: Optional[Any] = ..., **kwargs: Any): ...
+    async def wrap_key(self, vault_base_url: Any, key_name: Any, key_version: Any, algorithm: Any, value: Any, *, cls: Optional[Any] = ..., **kwargs: Any): ...
+    async def unwrap_key(self, vault_base_url: Any, key_name: Any, key_version: Any, algorithm: Any, value: Any, *, cls: Optional[Any] = ..., **kwargs: Any): ...
+    def get_deleted_keys(self, vault_base_url: Any, maxresults: Optional[Any] = ..., *, cls: Optional[Any] = ..., **kwargs: Any): ...
+    async def get_deleted_key(self, vault_base_url: Any, key_name: Any, *, cls: Optional[Any] = ..., **kwargs: Any): ...
```

Comment:
Is the use-case for 'cls' that the user is passing different callbacks into different methods? It seems like this would make sense as a 'client' level parameter instead of requiring it to be passed into each method.

For instance, I could see wanting to have it at the class level so you can get transparent tracing or logging (ie, factory method, create your client, etc..). Without this the consumer of the class has to carry along some callable just so it can pass it each time.

I have the same complaint about vault_base_url having to be passed into each method.